frontend/src/metabase/query_builder/actions/core/updateQuestion.ts

```diff
@@ -227,8 +227,16 @@ export const updateQuestion = (
       }
     }
 
-    const currentDependencies = currentQuestion?.query().dependentMetadata();
-    const nextDependencies = newQuestion.query().dependentMetadata();
+    const currentDependencies = currentQuestion
+      ? [
+          ...currentQuestion.dependentMetadata(),
+          ...currentQuestion.query().dependentMetadata(),
+        ]
+      : [];
+    const nextDependencies = [
+      ...newQuestion.dependentMetadata(),
+      ...newQuestion.query().dependentMetadata(),
+    ];
```

Comment:
Would it make sense to extend `Question.prototype.dependentMetadata` so it calls the underlying query's `dependentMetadata` method and merges the result with question-level metadata? Doesn't seem like we have any scenario when we either need query's metadata or question metadata separately anymore

Reply:
Yeah that makes sense. Let's punt and I'll make that change in a second PR.